src/electronic-ids/ms-cryptoapi/MsCryptoApiElectronicID.hpp

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright (c) 2022 Estonian Information System Authority
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+#pragma once
+
+#include "electronic-id/electronic-id.hpp"
+// TODO: move these to .cpp
+#include "pcsc-cpp/pcsc-cpp-utils.hpp"
+#include "../common.hpp"
+
+#include <windows.h>
+#include <wincrypt.h>
+
+namespace electronic_id
+{
+
+class MsCryptoApiElectronicID : public ElectronicID
+{
+public:
+    MsCryptoApiElectronicID(PCCERT_CONTEXT certCtx, pcsc_cpp::byte_vector&& cert,
+                            CertificateType cType, bool isRsa, HCRYPTPROV_OR_NCRYPT_KEY_HANDLE k,
+                            bool freeK) :
+        ElectronicID {std::make_unique<pcsc_cpp::SmartCard>()},
+        certContext {certCtx}, certData {cert}, certType {cType},
+        // TODO: SignatureAlgorithm::PS?
+        signatureAlgo {isRsa ? SignatureAlgorithm::RS : SignatureAlgorithm::ES}, key {k}, freeKey {
+                                                                                              freeK}
+    {
+    }
+
+    virtual ~MsCryptoApiElectronicID()
```

Comment:
```suggestion
   ~MsCryptoApiElectronicID()
```
Missing copy&move constructors and operators

Reply:
Thanks, good point!